src/cascadia/TerminalSettingsModel/FileUtils.cpp

```diff
@@ -123,15 +123,41 @@ namespace Microsoft::Terminal::Settings::Model
 
     void WriteUTF8FileAtomic(const std::filesystem::path& path, const std::string_view content)
     {
-        auto tmpPath = path;
+        // GH#10787: rename() will replace symbolic links themselves and not the path they point at.
+        // It's thus important that we first resolve them before generating temporary path.
+        std::error_code ec;
+        auto resolvedPath = std::filesystem::canonical(path, ec);
+        if (ec)
+        {
+            if (ec.value() != ERROR_FILE_NOT_FOUND)
+            {
+                THROW_WIN32_MSG(ec.value(), "failed to compute canonical path");
+            }
+
+            // We were not able to compute the canonical path as the target does not exist.
+            if (std::filesystem::is_symlink(path))

```

Comment:
As I said, IMO this check is unnecessary. We have various other fallbacks in place, in case a prior write corrupted the settings, so the most basic solution is IMO acceptable.
On the other hand this code is fine as it is. --> ✔️